### PR TITLE
Improve editor layout

### PR DESCRIPTION
**Pull request by jasonLaster**
Associated Issue: #891, #949

### Summary of Changes

- The footer no longer overlays the editor
- The footer is hidden when there is nothing to show
- Remove eroneous things
- Don't let users pretty print pretty files or bundle files

### CODE REVIEW

**Comment by wldcordeiro** [comment]
I'm closing my PR in favor of this. CodeMirror has a lot of layout quirks that make it a terrible fit for flexbox. 🙁

**Comment by jasonLaster** [comment]
@wldcordeiro  haha - sorry, i should have clarified. I was still hoping that you would come up w. a great flexbox fix, but didn't want to block on it :) 

Thanks for the hard work there

File: public/js/utils/editor.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+const { isPretty } = require("./source");
```
Comment (jlongster) [inline public/js/utils/editor.js]: What' the point of creating another file? I don't see why we should be splitting out components into multiple files between `utils` and `components`. It's a lot hard to follow the codebase.

File: public/js/utils/source-map-util.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ function generatedToOriginalId(generatedId, url) {
 }
 
 function isOriginalId(id) {
-  return id.match(/\/originalSource/);
+  return id && id.match(/\/originalSource/);
```
Comment (jlongster) [inline public/js/utils/source-map-util.js]: id's should definitely always exist. I'd rather not make things null-compatible; why would be pass in null? If you do `source.id` and it doesn't have an id that's a big problem. (this is also the reason why I don't like lodash. doing this makes bad code be silent)

**Comment by jlongster** [comment]
Sweet, thanks for all these fixes. A few nits that I'd like to discuss though.

File: public/js/utils/editor.js (hunk 1 above)
Comment (jasonLaster) [inline public/js/utils/editor.js]: unfortunately, both the footer and editor need to use this utility.

File: public/js/utils/source-map-util.js (hunk 2 above)
Comment (jasonLaster) [inline public/js/utils/source-map-util.js]: true - i'll look into this

**Comment by jlongster** [comment]
This PR still has problems:

1. The footer scrolls with the source. It's no longer "pinned" at the bottom. You should be able to repro by just loading a large source and scrolling.

screen shot 2016-10-24 at 10 03 47 am

2. The editor no longer is always 100% height. If the contents don't fill the screen, the editor is too short. It might be hard to tell, but you can see that the background color below the editor is gray, not white:

screen shot 2016-10-24 at 10 05 08 am

I just pulled the latest branch, I think, but if you can't repro maybe I did something wrong.

We need to fix this layout ASAP. There are other serious problems with the latest debugger in nightly, see https://bugzilla.mozilla.org/show_bug.cgi?id=1310330. It seems to be related to the functionality of "hiding" tabs when there isn't enough room, and it seems completely broken. The menu for listing hidden tabs is gone, and when it hides tabs, the entire area becomes scrollable.
